bson/value.go

```diff
@@ -553,6 +553,16 @@ func (v *Value) Binary() (subtype byte, data []byte) {
 	return st, b
 }
 
+// BinaryOK is the same as Binary, except it returns a boolean instead of
+// panicking.
+func (v *Value) BinaryOK() (subtype byte, data []byte, bool) {
+	if v == nil || v.offset == 0 || v.data == nil || v.data[v.start] != '\x05' {
```

Comment:
Let's use the `TypeBinary` here to make it consistent with the other *OK methods.